REM-1298-1299: implement get list of the transactions and fetch particular transaction (#18)
Remmeauth/remme-core-cli: README.md

368440.0
```

> ### Transactions

Reviewer: ```suggestion
### Transaction
```
Change everywhere, please.

Author: @anastasiia-bilova, done.

Get list of transactions — ``remme transactions get-list``:

| Arguments   | Type   |  Required | Description                                            |
| :---------: | :----: | :-------: | -----------------------------------------------------  |

> | ids         | String |  No       | List transaction by its identifier.                    |

Reviewer: ```suggestion
| ids         | String |  No       | Identifiers to get a list of transactions by.                |
```

Author: @anastasiia-bilova, done.

| start       | String |  No       | Id to start paging (inclusive).                        |
| limit       | Integer|  No       | Number of transaction to return.                       |
| head        | String |  No       | Id of head block.                                      |
| reverse     | String |  No       | Reverse result.                                        |
| node-url    | String |  No       | Apply the command to the specified node by its URL.    |
| family-name | String |  No       | List transaction by its family name.                   |

```bash
$ remme transactions get-list \

>     --limit=3 \

Reviewer: Make the parameters in the correct sequence like:
```python
'ids': transaction_ids,
'start': start,
'limit': limit,
'head': head,
'reverse': reverse,
'family_name': family_name,
```

Author: @anastasiia-bilova, done.

    --ids=13e46c3b07848b3a38be301b13a0a0b1f73b53766345b64eea1746031bb9672f5e6a34e123f56ce2a1af8523741d996c6f4c3cbf7bd75511547bcc4297d3c5cc \
    --start=13e46c3b07848b3a38be301b13a0a0b1f73b53766345b64eea1746031bb9672f5e6a34e123f56ce2a1af8523741d996c6f4c3cbf7bd75511547bcc4297d3c5cc \
    --head=792bef7ace0c13e47ddc28328952ed0494d28ecabacec961808b2c7bf93f765f5df2e8e32d1487fdb2d24b5cd0cde0ea2902a019243b9917793d3c26e834d8cf \
    --family_name=account
{
   "head": "792bef7ace0c13e47ddc28328952ed0494d28ecabacec961808b2c7bf93f765f
            5df2e8e32d1487fdb2d24b5cd0cde0ea2902a019243b9917793d3c26e834d8cf",
   "paging": {
      "limit": 3,
      "start": "13e46c3b07848b3a38be301b13a0a0b1f73b53766345b64eea1746031bb9672f
                5e6a34e123f56ce2a1af8523741d996c6f4c3cbf7bd75511547bcc4297d3c5cc",
      "next": ""
   },
   "data": []
}
```

Get single transaction by id — ``remme transactions get-single``:

| Arguments   | Type   |  Required | Description                                            |
| :---------: | :----: | :-------: | -----------------------------------------------------  |

> | identifier  | String |  Yes      | Transaction by its identifier.                         |

Reviewer: ```suggestion
| identifier  | String |  Yes      | Identifier to get transaction by.                         |
```

Author: @anastasiia-bilova, done.

368440.0
```

### Transaction

Get list of transaction — ``remme transaction get-list``:

| Arguments   | Type   |  Required | Description                                            |
| :---------: | :----: | :-------: | -----------------------------------------------------  |
| ids         | String |  No       | Identifiers to get a list of transaction by.          |

> | start       | String |  No       | Id to start paging (inclusive).                        |

Reviewer 2: `Id` > `identifier`.

Author: @dmytrostriletskyi, done.

> | limit       | Integer|  No       | Number of transaction to return.                       |

Reviewer 2: `transaction` > `transactions`.

Author: @dmytrostriletskyi, done.

> | head        | String |  No       | Id of head block.                                      |

Reviewer 2: `Id of head block` > `Identifier of block's head`.

Author: @dmytrostriletskyi, done.

> | reverse     | String |  No       | Reverse result.                                        |

Reviewer 2: `Reverse result` > `If transactions should be reversed`.

Author: @dmytrostriletskyi, done.

| node-url    | String |  No       | Node URL to apply a command to.                        |

> | family-name | String |  No       | List transaction by its family name.                   |

Reviewer 2: Could I specify only one family name or a few?

Author: Only one, because of the family name in core its [String field](https://github.com/Remmeauth/remme-core/blob/dev/remme/shared/forms/transaction.py#L24).

```bash
$ remme transaction get-list \

>     --ids=13e46c3b07848b3a38be301b13a0a0b1f73b53766345b64eea1746031bb9672f5e6a34e123f56ce2a1af8523741d996c6f4c3cbf7bd75511547bcc4297d3c5cc \

Reviewer 2: Could you show me how you enter multiple identifiers?

Author: @dmytrostriletskyi, 
`remme transaction get-list --ids='13e46c3b07848b3a38be301b13a0a0b1f73b53766345b64eea1746031bb9672f5e6a34e123f56ce2a1af8523741d996c6f4c3cbf7bd75511547bcc4297d3c5cc 1cb0b2af3c576551fe44bf44e1466f3eee1a2753c5038cb30aae56e47033804e3301a42fe14dccc8c870f07fac58934ac0bc4161f1d5249bdda0ea1d7d0b998f'`

    --start=13e46c3b07848b3a38be301b13a0a0b1f73b53766345b64eea1746031bb9672f5e6a34e123f56ce2a1af8523741d996c6f4c3cbf7bd75511547bcc4297d3c5cc \
    --limit=3 \
    --head=792bef7ace0c13e47ddc28328952ed0494d28ecabacec961808b2c7bf93f765f5df2e8e32d1487fdb2d24b5cd0cde0ea2902a019243b9917793d3c26e834d8cf \
    --reverse=false \
    --family_name=account
{
   "head": "792bef7ace0c13e47ddc28328952ed0494d28ecabacec961808b2c7bf93f765f
            5df2e8e32d1487fdb2d24b5cd0cde0ea2902a019243b9917793d3c26e834d8cf",
   "paging": {
      "limit": 3,
      "start": "13e46c3b07848b3a38be301b13a0a0b1f73b53766345b64eea1746031bb9672f
                5e6a34e123f56ce2a1af8523741d996c6f4c3cbf7bd75511547bcc4297d3c5cc",
      "next": ""
   },
   "data": []
}
```

> Get single transaction by id — ``remme transaction get-single``:

Reviewer 2: `id` > `identifier`.

Author: @dmytrostriletskyi, done.

}
```

### Transaction

> Get list of transaction — ``remme transaction get-list``:

Reviewer 2: `Get list of transaction` > `Get a list of transactions`

Author: @dmytrostriletskyi, done.

| Arguments   | Type   |  Required | Description                                            |
| :---------: | :----: | :-------: | -----------------------------------------------------  |
| ids         | String |  No       | Identifiers to get a list of transactions by.          |
| start       | String |  No       | Identifier to start paging (inclusive).                |
| limit       | Integer|  No       | Maximum amount of transactions to return.              |
| head        | String |  No       | Identifier of block's head.                            |
| reverse     | Bool   |  No       | If transactions should be reversed.                    |
| node-url    | String |  No       | Node URL to apply a command to.                        |
| family-name | String |  No       | List transaction by its family name.                   |

```bash
$ remme transaction get-list \

>     --ids=64d032fbaae9bc59f9e5484ec6f52cbceef567923456039a26a1cfb8bc9ee2431ac2b5de43efce28ef11820a3734dab9fa56db57a1b2fbdc2323036cceeab6ab \

Reviewer 2: Show the example of multiple ids, please.

Author: @dmytrostriletskyi, done.

| limit       | Integer|  No       | Maximum amount of transactions to return.              |
| head        | String |  No       | Block identifier to get a list of transactions from.   |
| reverse     | Bool   |  No       | Parameter to reverse result.                           |
| node-url    | String |  No       | Node URL to apply a command to.                        |
| family-name | String |  No       | List transaction by its family name.                   |

```bash
$ remme transaction get-list \
    --ids='64d032fbaae9bc59f9e5484ec6f52cbceef567923456039a26a1cfb8bc9ee2431ac2b5de43efce28ef11820a3734dab9fa56db57a1b2fbdc2323036cceeab6ab, 
    4a20a44a29fdb2c289dd7b137038e3bf5746ba888c0ee9f3a478db30fbdeb9200bed7c4cb15f4396412a3646fd4afeca0ed7bc6ec63eee0c665e4e508e7ed582' \
    --start=64d032fbaae9bc59f9e5484ec6f52cbceef567923456039a26a1cfb8bc9ee2431ac2b5de43efce28ef11820a3734dab9fa56db57a1b2fbdc2323036cceeab6ab \
    --limit=3 \
    --head=0f6601c8dd031370891d7b8c923fc46859911880fb971a5535da709a27e30b8569cadfbfe8349a678f447e28d03c4c4656245fed7abed4295b309ce1f859efbd \
    --reverse \
    --family-name=account \
    --node-url=node-6-testnet.remme.io
{
    "result": {

>         "data": [],

Reviewer 2: Create an example with data, please. We want to give a user a full understanding of how commands works and its output.

Author: @dmytrostriletskyi, done.

        "head": "0f6601c8dd031370891d7b8c923fc46859911880fb971a5535da709a27e30b8569cadfbfe8349a678f447e28d03c4c4656245fed7abed4295b309ce1f859efbd",
        "paging": {
            "limit": 3,
            "next": "",
            "start": "64d032fbaae9bc59f9e5484ec6f52cbceef567923456039a26a1cfb8bc9ee2431ac2b5de43efce28ef11820a3734dab9fa56db57a1b2fbdc2323036cceeab6ab"
        }
    }
}
```

Get a transaction by identifier — ``remme transaction get``:

| Arguments   | Type   |  Required | Description                                            |

> | :---------: | :----: | :-------: | -----------------------------------------------------  |

Reviewer: ```suggestion
| :---------: | :----: | :-------: | --------------------------------  |
```

Author: @anastasiia-bilova, done.
